linera-storage-service/src/client.rs

```diff
@@ -106,6 +107,24 @@ impl ReadableKeyValueStore<ServiceStoreError> for ServiceStoreClientInternal {
         Ok(test)
     }
 
+    async fn contains_keys(&self, keys: Vec<Vec<u8>>) -> Result<Vec<bool>, ServiceStoreError> {
+        let mut full_keys = Vec::new();
+        for key in keys {
+            ensure!(key.len() <= MAX_KEY_SIZE, ServiceStoreError::KeyTooLong);
+            let mut full_key = self.namespace.clone();
+            full_key.extend(&key);
```

Comment:
Why is the namespace not forwarded to the underlying store?

Reply:
The `namespace` is always specific to the instance created in the `memory` since each `MemoryStore` instance is related to a specific instance in memory. That can be seen with the `connect` implementation.

However, for a storage-service implementation, we have to do something else and since the namespace is not supported by the storage-service, the encapsulating of the namespace is the right solution.